### PR TITLE
fix(sql): fixed "invalid column" error, which was caused by incorrect…

### LINKED ISSUE #865: InvalidColumnException when explicitly selecting column names on JOIN
**Describe the bug**
Cairo throws an exception when using `SELECT column_name` in `JOIN` queries

**To Reproduce**
Run the following SQL:

```sql
CREATE TABLE sensors (ID LONG, make STRING, city STRING);

INSERT INTO sensors
SELECT
    x ID, --increasing integer
    rnd_str('Eberle', 'Honeywell', 'Omron', 'United Automation', 'RS Pro') make,
    rnd_str('New York', 'Miami', 'Boston', 'Chicago', 'San Francisco') city
FROM long_sequence(10000) x;

CREATE TABLE readings
AS(
    SELECT
        x ID,
        timestamp_sequence(to_timestamp('2019-10-17T00:00:00', 'yyyy-MM-ddTHH:mm:ss'), rnd_long(1,10,2) * 100000L) ts,
        rnd_double(0)*8 + 15 temp,
        rnd_long(0, 10000, 0) sensorId
    FROM long_sequence(10000000) x)
TIMESTAMP(ts)
PARTITION BY MONTH;

-- Throws an exception
SELECT city, max(temp)
FROM readings
JOIN(
    SELECT ID sensId, city
    FROM sensors)
ON readings.sensorId = sensId;

-- Will execute without exceptions
SELECT *
FROM readings
JOIN(
    SELECT ID sensId, city
    FROM sensors)
ON readings.sensorId = sensId;
```

**Workaround**
Using an alias on the joined table view returns correct results:

```sql
SELECT a.city, max(temp)
FROM readings
JOIN(
    SELECT ID sensId, make, city
    FROM sensors) a
ON readings.sensorId = a.sensId;
```

**Stack trace:**

```log
2021-03-17T11:17:16.829144Z I i.q.g.SqlCompiler plan [q=`select-group-by city, max(temp) max from (select [temp, sensorId] from readings timestamp (ts) join select [city, ID sensId] from (select-choose [city] ID sensId, make, city from (select [city] from sensors)) _xQdbA1 on sensId = readings.sensorId)`, fd=42]
2021-03-17T11:17:16.829285Z E i.q.c.h.p.JsonQueryProcessorState [42] Uh-oh. Error!
io.questdb.cairo.sql.InvalidColumnException
	at io.questdb.cairo.sql.InvalidColumnException.<clinit>(InvalidColumnException.java:28)
	at io.questdb.cairo.sql.RecordMetadata.getColumnIndex(RecordMetadata.java:44)
	at io.questdb.griffin.SqlCodeGenerator.lookupColumnIndexesUsingVanillaNames(SqlCodeGenerator.java:2690)
	at io.questdb.griffin.SqlCodeGenerator.processJoinContext(SqlCodeGenerator.java:2700)
	at io.questdb.griffin.SqlCodeGenerator.generateJoins(SqlCodeGenerator.java:850)
	at io.questdb.griffin.SqlCodeGenerator.generateSelect(SqlCodeGenerator.java:1456)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery0(SqlCodeGenerator.java:1229)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery(SqlCodeGenerator.java:1218)
	at io.questdb.griffin.SqlCodeGenerator.generateSubQuery(SqlCodeGenerator.java:2165)
	at io.questdb.griffin.SqlCodeGenerator.generateSelectGroupBy(SqlCodeGenerator.java:1876)
	at io.questdb.griffin.SqlCodeGenerator.generateSelect(SqlCodeGenerator.java:1445)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery0(SqlCodeGenerator.java:1229)
	at io.questdb.griffin.SqlCodeGenerator.generateQuery(SqlCodeGenerator.java:1218)
	at io.questdb.griffin.SqlCodeGenerator.generate(SqlCodeGenerator.java:643)
	at io.questdb.griffin.SqlCompiler.generate(SqlCompiler.java:1597)
	at io.questdb.griffin.SqlCompiler.compileUsingModel(SqlCompiler.java:1314)
	at io.questdb.griffin.SqlCompiler.compile(SqlCompiler.java:247)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.compileQuery(JsonQueryProcessor.java:302)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.execute0(JsonQueryProcessor.java:141)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.onRequestComplete(JsonQueryProcessor.java:179)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientRecv(HttpConnectionContext.java:582)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientOperation(HttpConnectionContext.java:185)
	at io.questdb.cutlass.http.HttpServer$1.lambda$$0(HttpServer.java:90)
	at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:151)
	at io.questdb.cutlass.http.HttpServer$1.run(HttpServer.java:94)
	at io.questdb.mp.Worker.run(Worker.java:114)
```

**Environment:**
 - branch: master (eb88a66)

**Additional Context:**
SQL which demonstrates this issue is documented at https://questdb.io/docs/get-started/first-database/

### PR DESCRIPTION
**Pull request by bluestreak01**
… top-down column propagation

Closes #865

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|15|15|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlOptimiser.java [hunk 1]
```diff
@@ -1126,6 +1140,7 @@ private void emitColumnLiteralsTopDown(ObjList<QueryColumn> columns, QueryModel
         for (int i = 0, n = columns.size(); i < n; i++) {
             final QueryColumn qc = columns.getQuick(i);
             emitLiteralsTopDown(qc.getAst(), target);
+//            addTopDownColumn(qc.getAlias(), target);
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/SqlOptimiser.java:1143]: looks unnecessary

**Review by ideoma — APPROVED** [review]

File: core/src/main/java/io/questdb/griffin/SqlOptimiser.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlOptimiser.java:1143]: 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|15|15|100.00%|

**Pull request merged**